math/mathcore/test/testMathRandom.cxx

```diff
@@ -21,197 +23,277 @@
 
 using namespace ROOT::Math;
 
-const long long  NR = 1E6;
+const long long NR = 1E6;
 double minPValue = 1.E-3;
 
-bool showPlots = false; 
+bool showPlots = false;
 
+inline void fillBits(int64_t *v, uint64_t x)
+{
+   for (int i = 0; i < 64; ++i) {
+      v[i] += x & 1;
+      x >>= 1;
+   }
+}
+
+template <typename Engine>
+bool testInt()
+{
+   Engine engine;
+   bool ret = true;
+   static_assert((~0ULL) >> (64 - Engine::kNumberOfBits) == Engine::MaxInt(), "MaxInt inconsistent with kNumberOfBits");
+   static_assert(0 == Engine::MinInt(), "MinInt not 0");
+#ifdef _WIN64
+   static_assert(Engine::kNumberOfBits == _popcount64(Engine::MaxInt()), "MaxInt inconsistent with kNumberOfBits");
+#else
+   static_assert(Engine::kNumberOfBits == __builtin_popcountll(Engine::MaxInt()),
+                 "MaxInt inconsistent with kNumberOfBits");
+   static_assert(64 - Engine::kNumberOfBits == __builtin_clzll(Engine::MaxInt()),
+                 "MaxInt inconsistent with kNumberOfBits");
+#endif
```

Comment:
Do these tests provide additional coverage? You're already testing above that `(~0ULL) >> (64 - Engine::kNumberOfBits) == Engine::MaxInt()`. Counting that there are exactly `Engine::kNumberOfBits` set is a weaker check, no?

Reply:
just verify that the expression is indeed constexpr.
(in principle they could fail is MaxInt is not all-lower-bits-one, such as 42)